### PR TITLE
[PBM-776] replica set names mapping

### PR DESCRIPTION
**Pull request by defbin**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | passed | failed |
| Sharded psmdb 4.4 | passed | failed |
| Sharded psmdb 5.0 | failed | failed |
| New cluster psmdb 4.2 | passed | - |
| New cluster psmdb 5.0 | passed | - |
| New cluster psmdb 4.4 | passed | - |
| Non-sharded psmdb 4.2 | passed | failed |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.2 | passed | passed |
| Single-node psmdb 5.0 | failed | passed |
| Single-node psmdb 4.4 | passed | passed |
| Non-sharded psmdb 4.4 | passed | passed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | passed | failed |
| New cluster psmdb 4.2 | passed | - |
| Sharded psmdb 5.0 | failed | failed |
| Sharded psmdb 4.4 | failed | failed |
| New cluster psmdb 5.0 | passed | - |
| New cluster psmdb 4.4 | passed | - |
| Non-sharded psmdb 4.4 | passed | passed |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.2 | passed | passed |
| Non-sharded psmdb 4.2 | passed | failed |
| Single-node psmdb 4.4 | passed | passed |
| Single-node psmdb 5.0 | failed | passed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | failed | failed |
| New cluster psmdb 4.2 | passed | - |
| Sharded psmdb 4.4 | passed | failed |
| Sharded psmdb 5.0 | failed | failed |
| New cluster psmdb 4.4 | passed | - |
| New cluster psmdb 5.0 | passed | - |
| Non-sharded psmdb 4.2 | failed | failed |
| Non-sharded psmdb 5.0 | failed | passed |
| Non-sharded psmdb 4.4 | failed | passed |
| Single-node psmdb 4.4 | passed | passed |
| Single-node psmdb 4.2 | passed | passed |
| Single-node psmdb 5.0 | failed | passed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | failed | failed |
| New cluster psmdb 4.2 | passed | - |
| Sharded psmdb 4.4 | failed | failed |
| Sharded psmdb 5.0 | failed | failed |
| New cluster psmdb 4.4 | passed | - |
| New cluster psmdb 5.0 | passed | - |
| Non-sharded psmdb 4.2 | failed | passed |
| Non-sharded psmdb 4.4 | failed | passed |
| Single-node psmdb 4.2 | failed | passed |
| Single-node psmdb 5.0 | failed | passed |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.4 | failed | passed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | passed | failed |
| Sharded psmdb 5.0 | failed | failed |
| Sharded psmdb 4.4 | passed | failed |
| New cluster psmdb 4.2 | passed | - |
| New cluster psmdb 4.4 | passed | - |
| New cluster psmdb 5.0 | passed | - |
| Non-sharded psmdb 4.2 | passed | failed |
| Non-sharded psmdb 4.4 | passed | passed |
| Single-node psmdb 4.2 | passed | passed |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.4 | passed | passed |
| Single-node psmdb 5.0 | failed | passed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | failed | failed |
| Sharded psmdb 4.4 | failed | failed |
| New cluster psmdb 4.2 | passed | - |
| New cluster psmdb 4.4 | passed | - |
| New cluster psmdb 5.0 | passed | - |
| Sharded psmdb 5.0 | failed | failed |
| Non-sharded psmdb 4.2 | failed | failed |
| Non-sharded psmdb 5.0 | failed | failed |
| Single-node psmdb 4.2 | failed | failed |
| Non-sharded psmdb 4.4 | failed | failed |
| Single-node psmdb 4.4 | failed | failed |
| Single-node psmdb 5.0 | failed | failed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| New cluster psmdb 4.2 | failed | - |
| Sharded psmdb 4.2 | failed | failed |
| Sharded psmdb 5.0 | failed | failed |
| Sharded psmdb 4.4 | failed | failed |

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 5.0 | - | failed |
| Sharded psmdb 4.2 | - | failed |

File: cli/backup.go [hunk 1]
```diff
@@ -129,39 +129,40 @@ func waitForBcpStatus(ctx context.Context, cn *pbm.PBM, bcpName string) (err err
...
 // changed to pbm.StatusError with respective error text emitted. It doesn't change meta on
 // storage nor in DB (backup is ok, it just doesn't cluster), it is just "in-flight" changes
 // in given `bcps`.
-func bcpsMatchCluster(bcps []pbm.BackupMeta, shards []pbm.Shard, confsrv string) {
+func bcpsMatchCluster(bcps []pbm.BackupMeta, shards []pbm.Shard, confsrv string, mapRS pbm.RSMapFunc) {
 	sh := make(map[string]struct{}, len(shards))
-	for _, s := range shards {
-		sh[s.RS] = struct{}{}
+	for i := range shards {
+		sh[shards[i].RS] = struct{}{}
```
[the remark this answers is not in the document]
Comment (defbin) [reply cli/backup.go:144]:
```
	for _, s := range shards {
```
is like
```
	for i := range shards {
        s := shards[i]
```
avoiding the whole `pbm.Shard` struct coping. not a big deal here. just habit/reflex

File: cli/list.go [hunk 2]
```diff
@@ -262,8 +269,8 @@ func getPitrList(cn *pbm.PBM, size int, full bool) (ranges []pitrRange, rsRanges
 	}
 
 	sh := make(map[string]struct{}, len(shards))
-	for _, s := range shards {
-		sh[s.RS] = struct{}{}
+	for i := range shards {
+		sh[shards[i].RS] = struct{}{}
```
[the remark this answers is not in the document]
Comment (defbin) [reply cli/list.go:273]: https://github.com/percona/percona-backup-mongodb/pull/652#discussion_r855826677

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| New cluster psmdb 4.2 | passed | - |
| New cluster psmdb 4.4 | passed | - |
| Sharded psmdb 4.2 | passed | failed |
| Sharded psmdb 4.4 | passed | failed |
| New cluster psmdb 5.0 | passed | - |
| Sharded psmdb 5.0 | failed | failed |
| Non-sharded psmdb 4.4 | failed | passed |
| Non-sharded psmdb 4.2 | passed | failed |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.2 | passed | passed |
| Single-node psmdb 5.0 | failed | passed |
| Single-node psmdb 4.4 | passed | passed |

**Review by dAdAbird — APPROVED** [review]
LGTM.
Would be nice to have CI tests to cover this functionality. We can add it later though

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| Sharded psmdb 4.2 | passed | failed |
| Sharded psmdb 4.4 | passed | failed |
| Sharded psmdb 5.0 | failed | failed |
| New cluster psmdb 4.2 | failed | - |
| Non-sharded psmdb 4.2 | passed | failed |
| New cluster psmdb 5.0 | passed | - |
| Non-sharded psmdb 4.4 | passed | passed |
| Single-node psmdb 4.2 | passed | passed |
| New cluster psmdb 4.4 | passed | - |
| Non-sharded psmdb 5.0 | failed | passed |
| Single-node psmdb 4.4 | passed | passed |
| Single-node psmdb 5.0 | failed | passed |

**Pull request merged**